src/js/data/CosmosClient.ts

```diff
@@ -0,0 +1,28 @@
+import { request, RequestResponse } from "@dcos/http-service";
+import { Observable } from "rxjs";
+
+export interface PackageVersionsResults {
+  [key: string]: string;
+}
+
+export interface PackageVersionsResponse {
+  results: PackageVersionsResults;
+}
+
+export function fetchPackageVersions(
+  packageName: string
+): Observable<RequestResponse<PackageVersionsResponse>> {
+  return request("/package/list-versions", {
```

Comment:
talking about consistency, it seems like some clients are handling error, some not. 🤔 

https://github.com/dcos/dcos-ui/pull/3579/files#diff-71851b32b10f6c1d63d2962b2b6f6e20R15

Reply:
That's a good point, we don't have any particular guid or even a good example of a client. So I would say we're in the process of establishing those good practices. Having said that I believe handling errors is a good practice.

Reply:
So for SDK Plans we did retries, caught and threw errors from the client. I've been torn on if its better to do that in the client on in the data-layer. I do agree we should be consistent, but I'm torn on the better place to place that logic.

Reply:
I ended up moving the error handling to the client since it would be the best place for parsing the error message from the response.